src/libraries/System.Private.CoreLib/src/System/Array.Enumerators.cs

```diff
@@ -139,11 +38,12 @@ public bool MoveNext()
         {
             get
             {
-                if (_index < 0)
+                nint index = _index;
+                if (index < 0)
```

Comment:
Does it make sense to reorder these with success first (predicted taken) and failures later?

```cs
if ((nuint)index < (nuint)_array.LongLength)) { return _array.InternalGetValue(index); }
else if (not_yet_started) { /* ... */ }
else { /* ... */ }
```

Reply:
I have rearranged it so that the flow looks like the flow of the generic enumerator in this file.

I have not done the manual predicten taken part. That's a problem for the JIT to deal with.